SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/ToJSONVisitor.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2023 VMware, Inc.
+ * SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.dbsp.sqlCompiler.compiler.visitors;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.jsontype.impl.LaissezFaireSubTypeValidator;
+import org.dbsp.sqlCompiler.circuit.*;
+import org.dbsp.sqlCompiler.circuit.operator.DBSPOperator;
+import org.dbsp.sqlCompiler.ir.CircuitVisitor;
+import org.dbsp.util.IndentStream;
+import org.dbsp.util.Utilities;
+
+public class ToJSONVisitor extends CircuitVisitor {
+    protected final IndentStream builder;
+
+    public ToJSONVisitor(IndentStream builder) {
+        super(true);
+        this.builder = builder;
+    }
+
+    @Override
+    public boolean preorder(DBSPCircuit circuit) {
```

Comment:
You could use org.json instead of reinventing all this.

Reply:
Jackson is doing the heavy lifting

Reply:
No, I mean all the code to do pretty printing and produce the json string could be delegated to Jackson or org.json.